doozer

```diff
@@ -942,12 +942,44 @@ def images_push(runtime, tag, version_release, to_defaults, late_only, to, dry_r
     runtime.state[cmd] = dict(state.TEMPLATE_IMAGE)
     lstate = runtime.state[cmd]  # get local convenience copy
 
+    pre_step = None
+    if 'images:build' in runtime.state:
+        pre_step = runtime.state['images:build']
+
+    required = []
+    pre_failed = []
+
+    if pre_step:
+        for img, status in pre_step['images'].iteritems():
+            if isinstance(status, str):
+                img_obj = runtime.image_map[img]
+                pre_failed.append(img)
+                if img_obj.required:
+                    required.append[img]
+
     version_release_tuple = None
 
     if version_release:
         version_release_tuple = version_release.split('-')
         click.echo('Setting up to push: version={} release={}'.format(version_release_tuple[0], version_release_tuple[1]))
 
+    items = runtime.image_metas()
+
+    if required:
+        msg = 'The following images failed during the previous step and are required:\n{}'.format('\n'.join(required))
```

Comment:
Same as `lstate['msg']`?

Reply:
Hmm... yeah, not sure why I did that. Fixing...